### ISSUE TITLE
How to render TButton as button and not input

### ISSUE THREAD

**Issue by frkinta** [pradosoft/prado#617]
Hello,

This is not a real issue but here is the "only place" I can expect an answer.

`<com:TButton><i class="fa fa-send"></i> Send</com:TButton>`
render a button without text. 

Is it possible to have as output `<button>...</button>` and not `<input ... />`?

**Comment by ctrlaltca** [comment]
Sorry, right now it's not possible. A few of html5 new input types have been implemented in https://github.com/pradosoft/prado/commit/f5b91473afb9ebeb2148594d1b1d1eae200f438b#diff-fdd94501c5dcbc3f0c0284de6562db44R634 , but <button> is not one of them.
They will be implemented before 4.0 bets released officially.

**Comment by ctrlaltca** [comment]
I'll keep this open as a reminder, thank you.

**Comment by emkael** [comment]
What about the `ButtonTag` property of `TButton`?

`<com:TButton ButtonTag="Button"><i class="fa fa-send"></i></com:TButton>`
should do what OP wants to achieve.

**Comment by ctrlaltca** [comment]
it will render a `<input type="button">`, not a `<button>`

**Comment by natsimhan** [comment]
Why do not use TLinkButton ? The result it's not <button>, but as you use bootstrap :)
and <a> is better <input> for many situations (for me ^^)

`<com:TLinkButton CssClass="btn btn-primary"><i class="fa fa-send"></i> Send</com:TLinkButton>`

**Comment by LCSKJ** [comment]
I think I added a TButtonTag class some time ago to use within TButton which causes it to render either as `<input>` or `<button>`, so it is possbile to to do it like @emkael suggested.

If you combine it with TWebControlDecorator's `setPreContentsText()` you can inject a `<span>` or `<i>` with the desired glyphicon classes. To make it easier I would even make a skin file to reuse, so you don't have to create the markup every time in template.

It might look similiar to this:

bsbutton.skin
```
<com:TButton SkinID="bsbutton" CssClass="btn btn-default">
<prop:Decorator.PreContentsText>
  <i class="fa fa-send"></i>
</prop:Decorator.PreContentsText>
</com:TButton>
```
and on template, set the SkinID everywhere you want to use it
```<com:TButton SkinID="bsbutton" Text="Send" />```

I did not test it though, so don't nail me down ... ;-)

**Comment by frkinta** [comment]
`<com:TButton ButtonTag="Button"><i class="fa fa-send"></i></com:TButton>` 
Is working well for my need.

Thanks you for all.

**Comment by Airakath** [comment]
Personally I use								
`<com:TLinkButton` 
   ID="linkButtonAnnuler" 
   Text="<span class='glyphicon glyphicon-remove' aria-hidden='true'></span> <br> Annuler" 
   OnClick="linkButtonAnnuler_Clicked" 
   CssClass="btn btn-default btn-Annuler"
/>

**Comment by ctrlaltca** [comment]
So, my fault for not knowing that @LCSKJ already implemented this in 4a4dd411abb23243d46a43e92deb4d22f2d34b17.
I added a bit of documentation in 30c3662
